src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -128,6 +130,20 @@
 public class EventsFacade extends AbstractIsaacFacade {
   private static final Logger log = LoggerFactory.getLogger(EventsFacade.class);
 
+  private static final String EXCEPTION_MESSAGE_CONTENT_ERROR_RETRIEVING_BOOKING =
```

Comment:
Not sure about all these string constants in the files in which they're used. Do we have lots of this across the code base? It might be worth putting them all in a single constants file or properties file

Reply:
We do have a (couple of) constants classes.
However, I was under the impression that the preferred practice is to keep constants that are only used in one class with that class and avoid general constant files unless those constants are actually used in multiple places?
There were a few strings constants that appeared in multiple places and as such have been added to the general constants file: [src/main/java/uk/ac/cam/cl/dtg/isaac/api/Constants.java](https://github.com/isaaccomputerscience/isaac-api/pull/156/files#diff-c69911cf78859bdea00c850c77a04272168965aea248cc0ea98840f3e5fe5cc0). All the ones defined in the class are, as far as I could determine via search, only used in that class.
Is this incorrect?

Reply:
I think it's best to keep constants close to the code that uses them where they are programatic - stuff like the column index constants.
For written language strings that might want to be changed relatively frequently there's an argument for keeping them somewhere accessible, rather than having to search/dig through the code to find the right class to modify. A bit like having properties files for strings rather than defining them in code.

I'm not particularly doctrinaire about it though, it was just a suggestion. Feel free not to.

Reply:
It's reasonable enough and it shouldn't be particularly difficult to relocate them. 🤷